--- post.php	2011-02-02 16:21:27.000000000 +0100
+++ post.php	2011-02-12 11:51:23.013581952 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 if ($pun_user['g_read_board'] == '0')
@@ -59,6 +60,7 @@
 // Did someone just hit "Submit" or "Preview"?
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Make sure form_user is correct
 	if (($pun_user['is_guest'] && $_POST['form_user'] != 'Guest') || (!$pun_user['is_guest'] && $_POST['form_user'] != $pun_user['username']))
 		message($lang_common['Bad request']);
@@ -658,6 +660,7 @@
 
 ?>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="submit" value="<?php echo $lang_common['Submit'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="s" /> <input type="submit" name="preview" value="<?php echo $lang_post['Preview'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="p" /> <a href="javascript:history.go(-1)"><?php echo $lang_common['Go back'] ?></a></p>
 		</form>
 	</div>
--- register.php	2011-02-02 16:21:27.000000000 +0100
+++ register.php	2011-02-12 11:51:23.013581952 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 // If we are logged in, we shouldn't be here
@@ -65,6 +66,7 @@
 
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Check that someone from this IP didn't register a user within the last hour (DoS prevention)
 	$result = $db->query('SELECT 1 FROM '.$db->prefix.'users WHERE registration_ip=\''.get_remote_address().'\' AND registered>'.(time() - 3600)) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 
@@ -402,6 +404,7 @@
 					</div>
 				</fieldset>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="register" value="<?php echo $lang_register['Register'] ?>" /></p>
 		</form>
 	</div>
