diff -Nura fluxbb-1.5.2.orig/post.php fluxbb-1.5.2/post.php
--- fluxbb-1.5.2.orig/post.php	2013-01-10 17:40:29.000000000 +0100
+++ fluxbb-1.5.2/post.php	2013-01-13 15:49:46.744773725 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 if ($pun_user['g_read_board'] == '0')
@@ -59,6 +60,7 @@
 // Did someone just hit "Submit" or "Preview"?
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Flood protection
 	if (!isset($_POST['preview']) && $pun_user['last_post'] != '' && (time() - $pun_user['last_post']) < $pun_user['g_post_flood'])
 		$errors[] = sprintf($lang_post['Flood start'], $pun_user['g_post_flood'], $pun_user['g_post_flood'] - (time() - $pun_user['last_post']));
@@ -695,6 +697,7 @@
 
 ?>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="submit" value="<?php echo $lang_common['Submit'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="s" /> <input type="submit" name="preview" value="<?php echo $lang_post['Preview'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="p" /> <a href="javascript:history.go(-1)"><?php echo $lang_common['Go back'] ?></a></p>
 		</form>
 	</div>
diff -Nura fluxbb-1.5.2.orig/register.php fluxbb-1.5.2/register.php
--- fluxbb-1.5.2.orig/register.php	2013-01-10 17:40:29.000000000 +0100
+++ fluxbb-1.5.2/register.php	2013-01-13 15:49:46.744773725 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 // If we are logged in, we shouldn't be here
@@ -65,6 +66,7 @@
 
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Check that someone from this IP didn't register a user within the last hour (DoS prevention)
 	$result = $db->query('SELECT 1 FROM '.$db->prefix.'users WHERE registration_ip=\''.$db->escape(get_remote_address()).'\' AND registered>'.(time() - 3600)) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 
@@ -427,6 +429,7 @@
 					</div>
 				</fieldset>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="register" value="<?php echo $lang_register['Register'] ?>" /></p>
 		</form>
 	</div>
diff -Nura fluxbb-1.5.2.orig/viewtopic.php fluxbb-1.5.2/viewtopic.php
--- fluxbb-1.5.2.orig/viewtopic.php	2013-01-10 17:40:29.000000000 +0100
+++ fluxbb-1.5.2/viewtopic.php	2013-01-13 15:49:46.744773725 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 if ($pun_user['g_read_board'] == '0')
@@ -442,6 +443,7 @@
 					</div>
 				</fieldset>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="submit" tabindex="<?php echo $cur_index++ ?>" value="<?php echo $lang_common['Submit'] ?>" accesskey="s" /> <input type="submit" name="preview" value="<?php echo $lang_topic['Preview'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="p" /></p>
 		</form>
 	</div>
