--- post.php	2014-01-09 16:47:04.109798856 +0100
+++ post.php	2014-01-09 05:42:34.785300955 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 if ($pun_user['g_read_board'] == '0')
@@ -59,6 +60,7 @@
 // Did someone just hit "Submit" or "Preview"?
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Flood protection
 	if (!isset($_POST['preview']) && $pun_user['last_post'] != '' && (time() - $pun_user['last_post']) < $pun_user['g_post_flood'])
 		$errors[] = sprintf($lang_post['Flood start'], $pun_user['g_post_flood'], $pun_user['g_post_flood'] - (time() - $pun_user['last_post']));
@@ -700,6 +702,7 @@
 
 ?>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="submit" value="<?php echo $lang_common['Submit'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="s" /> <input type="submit" name="preview" value="<?php echo $lang_post['Preview'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="p" /> <a href="javascript:history.go(-1)"><?php echo $lang_common['Go back'] ?></a></p>
 		</form>
 	</div>
--- register.php	2014-01-09 16:47:04.109798856 +0100
+++ register.php	2013-11-18 14:53:39.275533893 +0100
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 // If we are logged in, we shouldn't be here
@@ -65,6 +66,7 @@
 
 if (isset($_POST['form_sent']))
 {
+	check_funnyquestion() || $errors[] = $lang_funnyquestion['wrong-answer'];
 	// Check that someone from this IP didn't register a user within the last hour (DoS prevention)
 	$result = $db->query('SELECT 1 FROM '.$db->prefix.'users WHERE registration_ip=\''.$db->escape(get_remote_address()).'\' AND registered>'.(time() - 3600)) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 
@@ -430,6 +432,7 @@
 					</div>
 				</fieldset>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="register" value="<?php echo $lang_register['Register'] ?>" /></p>
 		</form>
 	</div>
--- viewtopic.php	2014-01-09 16:47:04.109798856 +0100
+++ viewtopic.php	2013-08-15 18:32:03.895427347 +0200
@@ -8,6 +8,7 @@
 
 define('PUN_ROOT', dirname(__FILE__).'/');
 require PUN_ROOT.'include/common.php';
+require PUN_ROOT.'include/funnyquestion.php';
 
 
 if ($pun_user['g_read_board'] == '0')
@@ -447,6 +448,7 @@
 					</div>
 				</fieldset>
 			</div>
+			<?php echo get_funnyquestion(); ?>
 			<p class="buttons"><input type="submit" name="submit" tabindex="<?php echo $cur_index++ ?>" value="<?php echo $lang_common['Submit'] ?>" accesskey="s" /> <input type="submit" name="preview" value="<?php echo $lang_topic['Preview'] ?>" tabindex="<?php echo $cur_index++ ?>" accesskey="p" /></p>
 		</form>
 	</div>
